cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -364,29 +377,41 @@ public void deleteSchedules(Iterable<? extends ScheduleId> scheduleIds) throws N
   @Override
   public void deleteSchedules(ApplicationId appId) {
     checkStarted();
-    execute((StoreAndQueueTxRunnable<Void, RuntimeException>) (store, queue) -> {
-      long deleteTime = System.currentTimeMillis();
-      deleteSchedulesInScheduler(store.listSchedules(appId));
-      List<ScheduleId> deleted = store.deleteSchedules(appId);
-      for (ScheduleId scheduleId : deleted) {
-        queue.markJobsForDeletion(scheduleId, deleteTime);
-      }
-      return null;
-    }, RuntimeException.class);
+    List<ScheduleId> deletedSchedules =
+      execute((StoreAndQueueTxRunnable<List<ScheduleId>, RuntimeException>) (store, queue) -> {
+        long deleteTime = System.currentTimeMillis();
+        List<ProgramSchedule> schedules = store.listSchedules(appId);
+        deleteSchedulesInScheduler(schedules);
+        List<ScheduleId> deleted = store.deleteSchedules(appId);
+        for (ScheduleId scheduleId : deleted) {
+          queue.markJobsForDeletion(scheduleId, deleteTime);
+        }
+        for (ProgramSchedule programSchedule : schedules) {
+          adminEventPublisher.publishScheduleDeletion(messagingContext, programSchedule.getScheduleId(),
+                                                      programSchedule);
+        }
+        return deleted;
+      }, RuntimeException.class);
   }
 
   @Override
   public void deleteSchedules(ProgramId programId) {
     checkStarted();
-    execute((StoreAndQueueTxRunnable<Void, RuntimeException>) (store, queue) -> {
-      long deleteTime = System.currentTimeMillis();
-      deleteSchedulesInScheduler(store.listSchedules(programId));
-      List<ScheduleId> deleted = store.deleteSchedules(programId);
-      for (ScheduleId scheduleId : deleted) {
-        queue.markJobsForDeletion(scheduleId, deleteTime);
-      }
-      return null;
-    }, RuntimeException.class);
+    List<ScheduleId> deletedSchedules =
+      execute((StoreAndQueueTxRunnable<List<ScheduleId>, RuntimeException>) (store, queue) -> {
+        long deleteTime = System.currentTimeMillis();
+        List<ProgramSchedule> schedules = store.listSchedules(programId);
+        deleteSchedulesInScheduler(schedules);
+        List<ScheduleId> deleted = store.deleteSchedules(programId);
+        for (ScheduleId scheduleId : deleted) {
+          queue.markJobsForDeletion(scheduleId, deleteTime);
+        }
+        for (ProgramSchedule programSchedule : schedules) {
```

Comment:
same question about deleted and schedules lists

Reply:
Confirmed with Mao, it should be same

Reply:
similar comment here

Reply:
Same as above